spec/rspec/expectations/failure_aggregator_spec.rb

```diff
@@ -265,6 +265,24 @@ def expect_error_included_in_aggregated_failure(error)
       end
     end
 
+    describe 'with MockExpectationError' do
+      it 'does not allow modifications of the failure list in the test example' do
+        expect {
+          aggregate_failures do
+            dbl = double
+            allow(double).to receive(:get).with(1) { [] }
+            array = dbl.get(2)
+            array << { :foo => :bar }
```

Comment:
> you'll get a weird message:

          NoMethodError:
            undefined method `upcase' for [#<RSpec::Mocks::MockExpectationError: #<Double (anonymous)> 

That's what happens now and what I've seen in my real-world test runs. It's surprising indeed ;-).

It's a tricky issue. I did considered returning `nil` but I wanted to make a minimal change the original implementation. I think there is no good way to solve it. We can't return the object specified by `and_return` because the expectation is not met. Whatever else we return, we'll get `NoMethodError` sooner or later (because we're returning a wrong object).

All I can think of is making the error message less surprising. Maybe we could return a special object that would raise an informative message from its `method_missing`, sth like `Previous mock expectation wasn't met. This object is only a dummy replacement`. WDYT?

Reply:
Strongly agree.
Since `array <<`/`string.upcase` are more likely to be happening in the SUT, and the chain of calls can be rather long, I'd prefer to fail early. And `nil` seems like a good tradeoff in terms of the clarity of the message and complexity of the handling.

Reply:
OK, replacing with `nil` then